ui/src/app/pages/signed-in/component.ts

```diff
@@ -89,25 +93,22 @@ export class SignedInComponent implements OnInit, OnDestroy, AfterViewInit {
     this.closeInactivityModal = this.closeInactivityModal.bind(this);
   }
 
-  ngOnInit(): void {
-    this.serverConfigService.getConfig().subscribe((config) => {
-      this.profileLoadingSub = this.profileStorageService.profile$.subscribe((profile) => {
-        this.profile = profile as unknown as FetchProfile;
-        setInstitutionCategoryState(this.profile.verifiedInstitutionalAffiliation);
-        if (hasRegisteredAccess(this.profile.accessTierShortNames)) {
-          cdrVersionsApi().getCdrVersions().then(resp => {
-            // cdrVersionsInitialized blocks app rendering so that route
-            // components don't try to lookup CDR data before it's available.
-            // This will need to be a step in the React bootstrapping as well.
-            // See discussion on https://github.com/all-of-us/workbench/pull/4713
-            cdrVersionStore.set(resp);
-            this.cdrVersionsInitialized = true;
-          });
-        } else {
-          this.cdrVersionsInitialized = true;
-        }
+  async ngOnInit() {
+    // We want to block app rendering on the presence of server config
+    // data so that route components don't try to lookup config data
+    // before it's available.
+    // This will need to be a step in the React bootstrapping as well.
+    // See discussion on https://github.com/all-of-us/workbench/pull/4713
+    if (serverConfigStore.get().config) {
```

Comment:
is there a case that's not covered by just replacing this `if/else` block with `serverConfigStore.subscribe(() => {}`?

Reply:
yes - `serverConfigStore.subscribe` only triggers on change of `serverConfigStore`. it doesn't trigger if there is something already in the store.